### ISSUE TITLE
BigQuery: False error of enclosing bracket

### ISSUE THREAD

**Issue by SuperBo** [sqlfluff/sqlfluff#900]
example query 
```sql
query = '''
  SELECT
    *
  FROM
    aa.table1 RIGHT JOIN aa.table2
    ON table1.a = table2.b AND (table1.a < table2.c)
```

This query is parsed fine when using dialect 'mysql' but got error when using 'bigquery' dialect.

## Expected Behaviour
No parse error

## Observed Behaviour
`SQLParseError("Found unexpected end bracket!, was expecting one of: [<OneOf: [<Ref: DotSegment>, <Sequence: [<Ref: DotSegment>]>]>, <OneOf: [<Ref: OnKeywordSegment>, <Ref: AsKeywordSegment>, <Ref: UsingKeywordSegment>, <Ref: CommaSegment>, <...]>, <sqlfluff.core.parser.grammar.noncode.NonCodeMatcher object at 0x7f3d35491d30>], but got <class 'sqlfluff.core.parser.segments.raw.end_bracket_SymbolSegment'>",)`

## Steps to Reproduce

```python
query = '''
    SELECT
        *
    FROM
        aa.table1 RIGHT JOIN aa.table2
    ON table1.a = table2.b AND (table1.a < table2.c)
'''
r = sqlfluff.parse(query, dialect='bigquery')
```

## Version
Include the output of `sqlfluff --version` along with your Python version
0.4.1
Python 3.6 and Python 3.9

**Comment by barrywhart** [comment]
I think this BigQuery bug is fixed in the master branch and the fix will be included in the upcoming release.

**Comment by barrywhart** [comment]
Confirmed. This is fixed in `master`.
